js/private/js_library_helpers.bzl

```diff
@@ -12,13 +12,8 @@ JS_LIBRARY_DATA_ATTR = attr.label_list(
     are added to the runfiles of this target. They should appear in the '*.runfiles' area of any executable which has
     a runtime dependency on this target.
 
-    If this list contains linked npm packages, npm package store targets or other targets that provide `JsInfo`,
-    `NpmPackageStoreInfo` providers are gathered from `JsInfo`. This is done directly from the
-    `npm_package_store_deps` field of these. For linked npm package targets, the underlying npm_package_store
-    target(s) that back the links is used.
-
-    Gathered `NpmPackageStoreInfo` providers are used downstream as direct dependencies when linking a downstream
-    `npm_package` target with `npm_link_package`.
+    Npm dependencies in `data` are propagated to the transitive npm dependencies of downstream `npm_package` targets when
```

Comment:
and those in `deps`?

Reply:
The difference being the ones in `deps` are also part of the ts compilation, otherwise both get passed along to the `npm_package` the same way for use at runtime. Correct?

Reply:
`data` is actually very similar to deps except that the `DefaultInfo` files of `data` are also included as inputs & passed along to runfiles. `data` allows files and doesn't require the target have a `JsInfo` provider.

Reply:
and only `deps` are included in the ts compilation, while both are passed along to runfiles...?

Reply:
correct; data is not available to the tsc action itself.

Reply:
Great, that all aligns with how I was thinking about it 👍 

Should `deps` also be mentioned here though? Also maybe the fact that `dev_deps` are not propagated?

Reply:
In `data`? I don't think so. The docstrings for `deps` and `dev_deps` will be right beside the docstring for `data`. This `JS_LIBRARY_DATA_ATTR` is just here because I re-use this docstring in rules_webpack and rules_esbuild

Reply:
the js_library `data` docstring uses this const as well:
```
"data": JS_LIBRARY_DATA_ATTR,
```

Reply:
> In data? 

Ah I see, I forgot what this comment thread was actually on 👍